lib/capistrano/tasks/migrations.rake

```diff
@@ -0,0 +1,123 @@
+namespace :forkcms do
+  namespace :migrations do
+    desc 'Prepare the server for running Fork CMS migrations'
+    task :prepare do
+      on roles(:web) do
+        create_migrations_file
+      end
+    end
+
+    desc 'Execute pending migrations'
+    task :execute do
+      on roles(:web) do
+        # Prepare for migrations if it's not done already
+        Rake::Task["forkcms:migrations:prepare"].invoke()
+
+        # If the current symlink doesn't exist yet, execute the first migration
+        execute_initial_migration if not test "[[ -e #{current_path} ]]"
+
+        # Abort if no migrations are found
+        migration_folders = get_migration_folders
+        if migration_folders.length == 0
+          next
+        end
+
+        # Fetch unprocessed migrations
+        migrations_to_execute = get_migrations_to_execute
+
+        # Abort if no new migrations are found
+        if migrations_to_execute.length == 0
+          next
+        end
+
+        # This can take a while and can go wrong. let's show a maintenance page
+        Rake::Task["forkcms:maintenance:enable"].invoke()
+
+        # Back up the database, just in case
+        Rake::Task["forkcms:database:backup"].invoke()
+
+        # Execute all migrations
+        migrations_to_execute.each do |dirname|
+          migration_path = "#{release_path}/migrations/#{dirname}"
+          migration_files = capture("ls -1 #{migration_path}").split(/\r?\n/)
+
+          migration_files.each do |filename|
+            # Update the locale through the console command
+            if filename.index('locale.xml') != nil
+              execute :php, "#{release_path}/bin/console forkcms:locale:import -f #{migration_path}/#{filename}"
+
+              next
+            end
+
+            # Execute a MySQL file
```

Comment:
Niet logischer om de MySQL eerst te doen.
In extremis kunnen er aanpassingen gedaan worden aan de Locale-table die nodig zijn om de import te laten slagen.

Reply:
Ik snap niet goed wa je bedoelt, kan je een voorbeeld geven?

Reply:
Stel u voor dat om de een of andere reden er een migration is die een kolom "foo" toevoeg in de tabel locale. En de code voor het importeren van de lokale rekent op die kolom.

Reply:
@tijsverkoyen wat als er in het mysql bestand een query staat die er van uit gaat dat de locale is aangevuld door een locale import?

Het is allebei zeer onwaarschijnlijk imo

Reply:
Mja, in mijn ogen moet ge eerst migrations doen, en dan pas console commands.

Reply:
89e1bd3